src/lib/uri/uri.h

```diff
@@ -13,6 +13,52 @@
 extern "C" {
 #endif /* defined(__cplusplus) */
 
+struct uri_option_value {
+        /** Each URI option value can be represented by string. */
+        char *value;
+};
+
+struct uri_option_values {
```

Comment:
Is there any point to have as a separate struct? Can't you inline it in the `uri_option` struct? I'd expect to see something like this:

````C
struct uri_option {
    char *name;
    int value_count;
    char *values[0];
};
```